### PR TITLE
VIH-9224 - Add VH SDS Staging URLS

### PR DESCRIPTION
**Pull request by chris-pateman**
### JIRA link (if applicable) ###
VIH-9224

### Change description ###
Add VH SDS Staging URLS

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by timja — COMMENTED** [review]
shouldn't this be in it's own zone?

**Comment by chris-pateman** [comment]
> shouldn't this be in it's own zone?
@timja 
There is only this zone for hearings.reform. We could request a new DNS zone but the process would take longer then we have.
The main aim of this is to demo the cut over for Video Hearing.

**Comment by timja** [comment]
> There is only this zone for hearings.reform. We could request a new DNS zone but the process would take longer then we have.

It's self service through this repo and would take you ~20 mins to setup?

Just create a new zone here, and once it's created point the NS records to the name servers of the zone e.g. https://github.com/hmcts/azure-public-dns/blob/master/environments/prod/platform-hmcts-net.yml#L2-L8

**Review by v1ferrarij — APPROVED** [review]
lgtm

**Comment by chris-pateman** [comment]
@timja like this?
*see new changes*

File: environments/staging/hearings-hmcts-net.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+ns: 
+  - name: "prod"
+    ttl: 900
+    record:
+      - "ns1-07.azure-dns.com"
+      - "ns2-07.azure-dns.net"
+      - "ns3-07.azure-dns.org"
+      - "ns4-07.azure-dns.info"
```
Comment (timja) [inline environments/staging/hearings-hmcts-net.yml:8]:
```suggestion
ns:  []
```

**Comment by timja** [comment]
> @timja like this? _see new changes_

almost

File: components/staging/hearings.tf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+data "local_file" "hearings-config" {
+  filename = "${path.cwd}/../../environments/staging/hearings-hmcts-net.yml"
+}
+
+module "hearings" {
+  source              = "../../modules/azure-public-dns/"
+  cname_records       = yamldecode(data.local_file.hearings-config.content).cname
+  a_recordsets        = yamldecode(data.local_file.hearings-config.content).A
+  srv_recordsets      = yamldecode(data.local_file.hearings-config.content).srv
+  zone_name           = "hearings.hmcts.net"
```
[the remark this answers is not in the document]
Comment (timja) [reply components/staging/hearings.tf:10]: do you want stg or staging?

File: environments/staging/hearings-hmcts-net.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  - name: "afdverify.video-stg"
+    ttl: 300
+    record: "afdverify.hmcts-stg.azurefd.net"
+  - name: admin-stg
+    ttl: 300
+    record: "hmcts-stg.azurefd.net"
+    shutter: false
+  - name: "afdverify.admin-stg"
+    ttl: 300
+    record: "afdverify.hmcts-stg.azurefd.net"
```
Comment (timja) [inline environments/staging/hearings-hmcts-net.yml:32]:
```suggestion
  - name: video
    ttl: 300
    record: "hmcts-stg.azurefd.net"
    shutter: false
  - name: "afdverify.video"
    ttl: 300
    record: "afdverify.hmcts-stg.azurefd.net"
  - name: admin
    ttl: 300
    record: "hmcts-stg.azurefd.net"
    shutter: false
  - name: "afdverify.admin"
    ttl: 300
    record: "afdverify.hmcts-stg.azurefd.net"
```

**Review by timja — APPROVED** [review]

File: components/staging/hearings.tf [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+module "hearings" {
+  source              = "../../modules/azure-public-dns/"
+  cname_records       = yamldecode(data.local_file.hearings-config.content).cname
+  a_recordsets        = yamldecode(data.local_file.hearings-config.content).A
+  srv_recordsets      = yamldecode(data.local_file.hearings-config.content).srv
+  zone_name           = "staging.hearings.hmcts.net"
+  resource_group_name = data.azurerm_resource_group.main.name
+  env                 = var.env
```
Comment (timja) [inline components/staging/hearings.tf:12]:
```suggestion
  env                 = "stg"
```

**Pull request merged**